packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,701 @@
+    _textDirection = value;
+    markNeedsLayout();
+  }
+
+  @override
+  void attach(PipelineOwner owner) {
+    super.attach(owner);
+    for (RenderBox child in _children) {
+      child.attach(owner);
+    }
+  }
+
+  @override
+  void detach() {
+    super.detach();
+    for (RenderBox child in _children) {
+      child.detach();
+    }
+  }
+
+  @override
+  void redepthChildren() {
+    _children.forEach(redepthChild);
+  }
+
+  @override
+  void visitChildren(RenderObjectVisitor visitor) {
+    _children.forEach(visitor);
+  }
+
+  static BoxParentData _boxParentData(RenderBox box) => box.parentData;
+
+  static Size _boxSize(RenderBox box) => box == null ? Size.zero : box.size;
+
+  static Rect _boxRect(RenderBox box) {
+    return box == null ? Rect.zero : _boxParentData(box).offset & box.size;
+  }
+
+  static double _minWidth(RenderBox box, double height) {
+    return box == null ? 0.0 : box.getMinIntrinsicWidth(height);
+  }
+
+  static double _minHeight(RenderBox box, double width) {
+    return box == null ? 0.0 : box.getMinIntrinsicHeight(width);
+  }
+
+  @override
+  double computeMinIntrinsicWidth(double height) {
+    double total = 0.0;
+    for (RenderBox box in boxes) {
+      total += _minWidth(box, height);
+    }
+    return total;
+  }
+
+  @override
+  double computeMaxIntrinsicWidth(double height) =>
+      computeMinIntrinsicWidth(height);
+
+  @override
+  double computeMinIntrinsicHeight(double width) =>
+      _minHeight(boxes.elementAt(0), width);
+
+  @override
+  double computeMaxIntrinsicHeight(double width) =>
+      computeMinIntrinsicHeight(width);
+
+  @override
+  double computeDistanceToActualBaseline(TextBaseline baseline) {
+    return boxes.elementAt(0).getDistanceToActualBaseline(baseline);
+  }
+
+  @override
+  void performLayout() {
+    double maxHeight = _kMinSegmentedControlHeight;
+    final Constraints parentConstraints = constraints.loosen();
+
+    for (RenderBox box in boxes) {
+      box.layout(parentConstraints, parentUsesSize: true);
+      final double boxHeight = box.size.height;
+      if (boxHeight > maxHeight) {
+        maxHeight = boxHeight;
+      }
+    }
+
+    final double overallWidth = constraints.maxWidth;
+    final Size overallSize =
+        constraints.constrain(Size(overallWidth, maxHeight));
+
+    Offset centerLayout(Size boxSize, double x) {
+      assert(maxHeight >= boxSize.height);
+      final Offset boxOffset =
+          new Offset(x, (maxHeight - boxSize.height) / 2.0);
+      return boxOffset;
+    }
+
+    final double childWidth = overallSize.width / boxes.length;
+
+    final BoxConstraints childConstraints =
+        new BoxConstraints.tightFor(width: childWidth, height: maxHeight);
+
+    for (RenderBox box in boxes) {
+      box.layout(childConstraints, parentUsesSize: true);
+    }
+
+    Offset childOffset = Offset.zero;
+
+    const double left = 0.0;
+
+    switch (textDirection) {
+      case TextDirection.rtl:
+        double start = left;
```

Comment:
Shouldn't this start on the right for RTL? Also, since it starts on the right, all the positions should be to the left of that.